posthog/api/test/test_person.py

```diff
@@ -88,44 +97,128 @@ def test_filter_by_cohort(self):
-        self.assertEqual(len(response["results"]), 1, response)
+        response = self.client.get(f"/api/person/?cohort={cohort.pk}")
+        self.assertEqual(len(response.json()["results"]), 1, response)
+
+    def test_filter_person_list(self):
+
+        person1: Person = Person.objects.create(
+            team=self.team, distinct_ids=["distinct_id", "another_one"], properties={"email": "someone@gmail.com"},
+        )
+        person2: Person = Person.objects.create(
+            team=self.team, distinct_ids=["distinct_id_2"], properties={"email": "another@gmail.com"},
+        )
+
+        # Filter by distinct ID
+        response = self.client.get("/api/person/?distinct_id=distinct_id")  # must be exact matches
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertEqual(len(response.json()["results"]), 1)
+        self.assertEqual(response.json()["results"][0]["id"], person1.pk)
+
+        response = self.client.get("/api/person/?distinct_id=another_one")  # can search on any of the distinct IDs
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertEqual(len(response.json()["results"]), 1)
+        self.assertEqual(response.json()["results"][0]["id"], person1.pk)
+
+        # Filter by email
+        response = self.client.get("/api/person/?email=another@gmail.com")
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertEqual(len(response.json()["results"]), 1)
+        self.assertEqual(response.json()["results"][0]["id"], person2.pk)
+
+        # Filter by key identifier
+        for _identifier in ["another@gmail.com", "distinct_id_2"]:
+            response = self.client.get(f"/api/person/?key_identifier={_identifier}")
+            self.assertEqual(response.status_code, status.HTTP_200_OK)
+            self.assertEqual(len(response.json()["results"]), 1)
+            self.assertEqual(response.json()["results"][0]["id"], person2.pk)
+
+        # Non-matches return an empty list
+        response = self.client.get("/api/person/?email=inexistent")
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertEqual(len(response.json()["results"]), 0)
+
+        response = self.client.get("/api/person/?distinct_id=inexistent")
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertEqual(len(response.json()["results"]), 0)
+
+    def test_cant_see_another_orgs_pii_with_filters(self):
+
+        another_org: Organization = Organization.objects.create()
+        another_team: Team = Team.objects.create(organization=self.organization)
+
+        another_person1: Person = Person.objects.create(
+            team=another_team, distinct_ids=["distinct_id", "x_another_one"],
+        )
+        another_person2: Person = Person.objects.create(
+            team=another_team, distinct_ids=["x_distinct_id_2"], properties={"email": "team2_another@gmail.com"},
+        )
```

Comment:
Nit: this test would be more on point if the different persons shared the same distinct_id for whatever reason (e.g. "user_1" could be a common id)

Reply:
I think it should be OK, but paging @paolodamico

Reply:
Actually this is already the case, `another_person1` shares a distinct ID ["distinct_id"] with `person` (which is a person from a different org) [line 157], to effectively test that even if a person in another team has the same ID as a person in your team, you won't see information from another team.